cedar-lean/Cedar/Validation/Validator.lean

```diff
@@ -51,6 +51,23 @@ def Schema.toEnvironments (schema : Schema) : List Environment :=
     reqty := ·
   })
 
+/-- Return the environment for the particular (p,a,r) tuple, or `none` if this
+is not a valid tuple in this schema -/
+def Schema.getEnvironment (schema : Schema) (principalTy resourceTy : EntityType) (action : EntityUID) : Option Environment := do
+  let ase ← schema.acts.find? action
+  match (ase.appliesToPrincipal.contains principalTy, ase.appliesToResource.contains resourceTy) with
```

Comment:
I suggest turning this into a parallel match instead of a tuple match:  proofs are easier on parallel matches (especially if there is any recursion involved).

```
match ase.appliesToPrincipal.contains principalTy, ase.appliesToResource.contains resourceTy with
| true, true => ...
| _ => ...
```